<a href="https://colab.research.google.com/github/dngg6688/LLM_Studycamp/blob/main/8.Finetuning%20Large%20Language%20Models/w8_l1_Why_finetuning_lab_student.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Compare finetuned vs. non-finetuned models

In [1]:
!pip install lamini

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 394.2/394.2 kB 7.6 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 193.4/193.4 kB 9.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 41.7/41.7 kB 5.9 MB/s eta 0:00:00


In [2]:
from google.colab import userdata
import lamini
lamini.api_key = userdata.get('Lamini_API_KEY')

# llm = lamini.Lamini("meta-llama/Llama-2-7b-chat-hf")
# print(llm.generate("How are you?"))

In [3]:
from llama import BasicModelRunner

### Try Non-Finetuned models

In [4]:
non_finetuned = BasicModelRunner("meta-llama/Llama-2-7b-hf")

In [5]:
non_finetuned_output = non_finetuned("Tell me how to train my dog to sit")

In [6]:
print(non_finetuned_output)

.
Tell me how to train my dog to sit. I have a 10 month old puppy and I want to train him to sit. I have tried the treat method and the verbal command method. I have tried both and he just doesn't seem to get it. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit by putting my hand on his back and pushing him down. I have tried to get him to sit 

In [7]:
print(non_finetuned("What do you think of Mars?"))


I think it's a great place to visit.
I think it's a great place to live.
I think it's a great place to die.
I think it's a great place to be born.
I think it's a great place to be.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.
I think it's a great place to be from.

In [8]:
print(non_finetuned("taylor swift's best friend"))


I'm not sure if I've ever mentioned this before, but I'm a huge Taylor Swift fan. I've been a fan since her first album, and I've been a fan ever since. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her music, her lyrics, her style, her personality, and her personality. I've been a fan of her musi

我不確定我以前是否提過這一點，但我是泰勒絲的超級粉絲。 從她的第一張專輯開始我就一直是她的粉絲，從那時起我就一直是她的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。 我一直是她的音樂、她的歌詞、她的風格、她的個性和她的個性的粉絲。...

In [9]:
print(non_finetuned("""Agent: I'm here to help you with your Amazon deliver order.
Customer: I didn't get my item
Agent: I'm sorry to hear that. Which item was it?
Customer: the blanket
Agent:"""))

 I'm sorry to hear that. I'll look into it.
Customer: I'm not sure if I got the right blanket.
Agent: I'm sorry to hear that. I'll look into it.
Customer: I'm not sure if I got the right blanket. I'm not sure if I got the right blanket.
Agent: I'm sorry to hear that. I'll look into it. I'll look into it.
Customer: I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket.
Agent: I'm sorry to hear that. I'll look into it. I'll look into it. I'll look into it.
Customer: I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure if I got the right blanket. I'm not sure

聽到這個消息我很遺憾。 我會調查一下。
顧客：我不確定我買的毯子是否合適。
特工： 聽到這個消息我很遺憾。 我會調查一下。
顧客：我不確定我買的毯子是否合適。 我不確定我是否買對了毯子。
特工： 聽到這個消息我很遺憾。 我會調查一下。 我會調查一下。
顧客：我不確定我買的毯子是否合適。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。
特工： 聽到這個消息我很遺憾。 我會調查一下。 我會調查一下。 我會調查一下。
顧客：我不確定我買的毯子是否合適。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。 我不確定我是否買對了毯子。

### Compare to finetuned models

In [10]:
finetuned_model = BasicModelRunner("meta-llama/Llama-2-7b-chat-hf")

In [11]:
finetuned_output = finetuned_model("Tell me how to train my dog to sit")

In [12]:
print(finetuned_output)

 on command.
Training a dog to sit on command is a basic obedience command that can be achieved with patience, consistency, and positive reinforcement. Here's a step-by-step guide on how to train your dog to sit on command:

1. Choose a quiet and distraction-free area: Find a quiet area with minimal distractions where your dog can focus on you.
2. Have treats ready: Choose your dog's favorite treats and have them ready to use as rewards.
3. Stand in front of your dog: Stand in front of your dog and hold a treat close to their nose.
4. Move the treat up and back: Slowly move the treat up and back, towards your dog's tail, while saying "sit" in a calm and clear voice.
5. Dog will sit: As you move the treat, your dog will naturally sit down to follow the treat. The moment their bottom touches the ground, say "good sit" and give them the treat.
6. Repeat the process: Repeat steps 3-5 several times, so your dog starts to associate the command "sit" with the action of sitting down.
7. Gradua

根據命令。
訓練狗狗按照命令坐下是一種基本的服從命令，可以透過耐心、一致性和積極強化來實現。 以下是有關如何訓練您的狗按照命令坐下的分步指南：

1. 選擇一個安靜且無幹擾的區域：找到一個幹擾最少的安靜區域，讓您的狗可以專注於您。
2. 準備好零食：選擇您的狗狗最喜歡的零食並準備好作為獎勵。
3. 站在你的狗面前：站在你的狗面前，把零食放在靠近狗鼻子的地方。
4. 向上和向後移動零食：慢慢地向上和向後移動零食，靠近狗的尾巴，同時用平靜而清晰的聲音說「坐下」。
5. 狗狗會坐下：當您移動零食時，您的狗狗會自然地坐下來跟隨零食。 當他們的屁股接觸地面時，說“坐好”並給他們零食。
6.重複這個過程：重複步驟3-5幾次，這樣你的狗就開始將「坐下」這個指令與坐下的動作連結起來。
7. 逐漸停止零食：當你的狗對命令變得更加適應時，開始逐步停止零食。 相反，用讚美和喜愛作為獎勵。
8. 練習、練習、再練習：在一天中的不同時間、不同的地點、不同的干擾下練習「坐」指令。 這將幫助您的狗了解該命令不僅僅是一個技巧，而是一個基本的服從命令。
9. 保持一致：訓練狗狗時，一致性是關鍵。 確保家裡的每個人都使用相同的命令和獎勵系統。
10.要有耐心：訓練狗狗需要時間和耐心。 如果您的狗沒有立即接受命令，請不要感到沮喪。 繼續練習，最終你的狗會學會的。

請記住，訓練狗是一個旅程，保持耐心、一致和積極非常重要。 經過時間和練習，您的狗將學會按照命令坐下和其他基本的服從命令。

In [13]:
print(finetuned_model("[INST]Tell me how to train my dog to sit[/INST]"))

  Training your dog to sit is a basic obedience command that can be achieved with patience, consistency, and positive reinforcement. Here's a step-by-step guide on how to train your dog to sit:

1. Choose a quiet and distraction-free area: Find a quiet area with no distractions where your dog can focus on you.
2. Have treats ready: Choose your dog's favorite treats and have them ready to use as rewards.
3. Stand in front of your dog: Stand in front of your dog and hold a treat close to their nose.
4. Move the treat up and back: Slowly move the treat up and back, towards your dog's tail, while saying "sit" in a calm and clear voice.
5. Dog will sit: As you move the treat, your dog will naturally sit down to follow the treat. The moment their bottom touches the ground, say "good sit" and give them the treat.
6. Repeat the process: Repeat steps 3-5 several times, so your dog starts to associate the command "sit" with the action of sitting down.
7. Gradually phase out the treats: As your d

訓練你的狗坐下是一個基本的服從命令，可以透過耐心、一致性和積極強化來實現。 以下是如何訓練您的狗狗坐下的分步指南：

1. 選擇一個安靜且不受干擾的區域：找到一個不受干擾的安靜區域，讓您的狗可以專注於您。
2. 準備好零食：選擇您的狗狗最喜歡的零食並準備好作為獎勵。
3. 站在你的狗面前：站在你的狗面前，把零食放在靠近狗鼻子的地方。
4. 向上和向後移動零食：慢慢地向上和向後移動零食，靠近狗的尾巴，同時用平靜而清晰的聲音說「坐下」。
5. 狗狗會坐下：當您移動零食時，您的狗狗會自然地坐下來跟隨零食。 當他們的屁股接觸地面時，說“坐好”並給他們零食。
6.重複這個過程：重複步驟3-5幾次，這樣你的狗就開始將「坐下」這個指令與坐下的動作連結起來。
7. 逐漸停止零食：當你的狗對命令變得更加適應時，開始逐步停止零食。 相反，用讚美和喜愛作為獎勵。
8. 練習、練習、再練習：在一天中的不同時間、不同的地點、不同的干擾下練習「坐」指令。 這將幫助您的狗了解該命令不僅僅是一個技巧，而是一個基本的服從命令。
9. 保持一致：訓練狗狗時，一致性是關鍵。 確保家裡的每個人都使用相同的命令和獎勵系統。
10.要有耐心：訓練狗狗需要時間和耐心。 如果您的狗沒有立即接受命令，請不要感到沮喪。 繼續練習，最終你的狗會學會的。

附加提示：

* 儘早開始訓練：越早開始訓練你的狗，它們就越容易學習。
* 使用一致的命令：為「坐下」選擇一致的命令，並在每次您希望您的狗坐下時使用它。
* 輕輕觸摸：使用指令時，輕輕觸摸狗狗的鼻子，引導它們進入坐姿。
* 清晰簡潔：使用清晰簡潔的命令，例如“坐下”或“坐下”，避免使用長句子或複雜的單字。
* 在不同情況下練習：在不同情況下練習「坐」命令，例如散步時、公園里或家裡，幫助您的狗理解該命令不僅僅是一個技巧，而是一個基本的服從命令。

請記住，訓練狗狗需要時間、耐​​心和一致性。 透過積極的強化和重複，您的狗很快就會學會「坐下」命令。

In [14]:
print(non_finetuned("[INST]Tell me how to train my dog to sit[/INST]"))


[INST]Tell me how to train my dog to sit[/INST]
[INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to sit[/INST] [INST]Tell me how to train my dog to si

In [15]:
print(finetuned_model("What do you think of Mars?"))



Mars is a fascinating planet that has captured the imagination of humans for centuries. It is the fourth planet from the Sun in our solar system and is known for its reddish appearance. Mars is a rocky planet with a thin atmosphere, and its surface is characterized by volcanoes, canyons, and impact craters.

One of the most intriguing aspects of Mars is its potential to support life. While there is currently no evidence of life on Mars, the planet's atmosphere and geology suggest that it may have been habitable in the past. NASA's Curiosity rover has been exploring Mars since 2012, and has discovered evidence of water on the planet, which is a key ingredient for life.

Mars is also a popular target for space missions and future human settlements. Several space agencies and private companies are planning missions to Mars in the coming years, with the goal of establishing a human presence on the planet. The challenges of establishing a human settlement on Mars are significant, includin

火星是一個迷人的星球，幾個世紀以來一直吸引著人類的想像。 它是太陽系中距離太陽第四遠的行星，以其微紅色的外觀而聞名。 火星是一顆岩石行星，大氣層稀薄，其表面以火山、峽谷和撞擊坑為特徵。

火星最有趣的方面之一是它支持生命的潛力。 雖然目前火星上沒有生命存在的證據，但該行星的大氣層和地質表明它過去可能適合居住。 美國太空總署的好奇號火星車自 2012 年以來一直在探索火星，並在火星上發現了水的證據，而水是生命的關鍵成分。

火星也是太空任務和未來人類定居點的熱門目標。 一些航太機構和私人公司正在計劃未來幾年的火星任務，目標是在地球上建立人類存在。 在火星上建立人類定居點面臨巨大的挑戰，包括惡劣的環境、缺乏資源以及距離地球的距離。 然而，許多專家認為，火星是探索太陽系和人類擴張的下一個合乎邏輯的步驟。

總之，火星是一個迷人而複雜的星球，不斷激發人類的想像。 雖然目前火星上沒有生命存在的證據，但該行星在過去和未來支持生命的潛力使其成為科學探索和未來人類定居的重要目標。

In [16]:
print(finetuned_model("taylor swift's best friend"))



Taylor Swift's best friend is a person who has been by her side through thick and thin. Here are some possible candidates:

1. Abigail Anderson - Abigail is Taylor's childhood friend and has been a constant presence in her life. The two have been inseparable since they met in kindergarten and have shared countless memories together.
2. Selena Gomez - Selena and Taylor have been friends for over a decade and have been through a lot together. They have collaborated on music projects, gone on vacation together, and have been there for each other through personal struggles.
3. Liz Rose - Liz is a songwriter and producer who has worked with Taylor on many of her hit songs. The two have a close professional relationship and have also become close friends over the years.
4. Joe Jonas - Joe and Taylor were in a relationship for a few years in the early 2000s and have remained close friends since then. They have been spotted together at various events and have even collaborated on music proje

泰勒絲最好的朋友是一個一直陪伴在她身邊的人。 以下是一些可能的候選人：

1.阿比蓋爾·安德森－阿比蓋爾是泰勒兒時的朋友，一直出現在她的生命中。 兩人從幼兒園相識起就形影不離，共同留下了無數回憶。

2.賽琳娜戈梅茲（Selena Gomez）－賽琳娜和泰勒已經是十多年的朋友了，一起經歷了很多事情。 他們在音樂項目上進行了合作，一起去度假，並在個人奮鬥中互相支持。

3.Liz Rose - Liz 是一位詞曲作者和製作人，曾與泰勒合作創作了許多熱門歌曲。 兩人有著密切的職業關係，多年來也成為了親密的朋友。

4.喬·喬納斯 (Joe Jonas) - 喬和泰勒在 2000 年代初期交往了幾年，從那時起一直是親密的朋友。 他們被發現一起出現在各種活動中，甚至在音樂項目上進行了合作。

5.Ed Sheeran - Ed 和 Taylor 是多年好友，並合作創作了多首歌曲。 他們有著密切的聯繫，並在個人奮鬥和勝利中相互支持。

總的來說，泰勒絲最好的朋友是她生活中不斷支持和鼓勵的人。 無論是兒時的朋友、音樂合作者或浪漫的伴侶，泰勒最好的朋友都一直陪伴在她身邊。

In [17]:
print(finetuned_model("""Agent: I'm here to help you with your Amazon deliver order.
Customer: I didn't get my item
Agent: I'm sorry to hear that. Which item was it?
Customer: the blanket
Agent:"""))

 I see. Can you please provide me with your order number so I can look into this for you?
Customer: I don't have the order number.
Agent: Okay, no worries. Can you please tell me the name of the item you didn't receive?
Customer: It was a blue blanket.
Agent: I see. I'm going to check on the status of your order for you. Can you please hold for just a moment?
Customer: Okay, thank you.
Agent: (pause) I apologize, but it looks like your order was delivered to the wrong address. Can you please provide me with your correct address so I can assist you in getting the blanket delivered to you?
Customer: (sighs) I don't know why this is happening to me. I've been ordering from Amazon for years and never had any problems.
Agent: I apologize for any inconvenience this has caused. I'm here to help you resolve the issue as quickly as possible. Can you please provide me with your correct address so I can assist you?
Customer: (frustrated) Fine. It's 123 Main St, Apartment 302. Can you please fix t

我懂了。 您能否提供我您的訂單編號，以便我幫您調查一下？

顧客：我沒有訂單號碼。

代理人：好的，不用擔心。 您能告訴我您未收到的物品的名稱嗎？

顧客：那是一條藍色的毯子。

代理：我明白了。 我將為您檢查您的訂單狀態。 你能稍等一下嗎？

顧客：好的，謝謝。

代理：（停頓）很抱歉，您的訂單似乎發送到了錯誤的地址。 您能否提供我您的正確地址，以便我幫助您將毯子送到您的手中？

顧客：（嘆氣）我不知道為什麼會發生這種事。 我從亞馬遜訂購已有多年，從未遇到任何問題。

代理：對於由此造成的任何不便，我深表歉意。 我來這裡是為了幫助您盡快解決問題。 您能否提供我您的正確地址，以便我為您提供協助？

顧客：（沮喪）好。 地址是 123 Main St，302 號公寓。您能修一下嗎？

代理：（點頭）當然，我會繼續更新您訂單上的地址。 在我處理這個問題時，您能稍等一下嗎？

顧客：（嘆氣）好的，謝謝。

代理：（停頓）太好了，我已經更新了您訂單上的地址。 您的毯子應該會在接下來的 3-5 個工作天內送到您的新地址。 還有什麼我可以幫助您的嗎？

顧客：（失望）不，就這樣了。 感謝您的協助。

代理人： 不客氣。 我還能為您做些什麼嗎？

顧客：（猶豫）不，我想就是這樣了。 再次感謝你。

代理人： 不客氣。 祝你有美好的一天！

在這種情況下，客戶服務代理正在嘗試幫助客戶解決亞馬遜送貨訂單的問題。 客服人員在整個對話過程中表現得彬彬有禮且專業，並向客戶提供解決問題所需的資訊。 客戶感到沮喪和失望，但客服人員在整個談話過程中仍然保持耐心並提供協助。

### Compare to ChatGPT

Tell me how to train my dog to sit

ChatGPT responds with: ChatGPT 回應：

Training your dog to sit is a basic and essential command that can be taught using positive reinforcement. Here's a simple step-by-step guide:

訓練你的狗坐下是一項基本和必不可少的命令，可以使用正強化來教授。以下是一個簡單的分步指南：

Prepare Treats: Gather small, soft treats that your dog enjoys. Make sure they are easy to chew and won't take too long to eat.

準備零食：收集您的狗喜歡的小而柔軟的零食。確保它們易於咀嚼並且不會花太長時間食用。

Find a Quiet Space: Choose a quiet area with minimal distractions for the training session. This will help your dog focus better.

尋找一個安靜的空間：為訓練課程選擇一個干擾最小的安靜區域。這將説明您的狗更好地集中注意力。

Get Your Dog's Attention: Call your dog's name to get their attention. Make sure they are looking at you.

引起狗的注意：叫你的狗的名字以引起他們的注意。確保他們正在看著你。

Use a Treat to Lure: Hold a treat close to your dog's nose, and slowly move your hand upward and slightly backward over their head. As you do this, your dog's natural response will be to follow the treat with their nose, causing them to sit.

使用零食引誘：將零食靠近狗的鼻子，然後慢慢地將手向上移動並稍微向後移過它們的頭頂。當你這樣做時，你的狗的自然反應將是用鼻子跟隨零食，讓它們坐下。

Say the Command: As your dog starts to sit, say the command "Sit" in a clear and firm voice. Use the word consistently every time you want your dog to sit.

說出命令：當你的狗開始坐下時，用清晰而堅定的聲音說出命令“坐下”。每次你想讓你的狗坐下時，都始終如一地使用這個詞。

Reward and Praise: As soon as your dog sits, immediately reward them with the treat and offer verbal praise. This positive reinforcement will help them associate sitting with positive outcomes.

獎勵和表揚：一旦您的狗坐下，立即用款待獎勵它們並提供口頭表揚。這種積極的強化將幫助他們將坐著與積極的結果聯繫起來。

Repeat and Practice: Repeat the process several times in a row during each training session. Keep the sessions short (around 5-10 minutes) to prevent your dog from losing interest.

重複和練習：在每次訓練期間連續多次重複該過程。保持簡短的會議（約 5-10 分鐘），以防止您的狗失去興趣。

Add Duration: Once your dog consistently sits on command, you can gradually increase the duration by waiting a couple of seconds before giving the treat. This helps reinforce the sit command.

添加持續時間：一旦您的狗始終如一地聽從命令，您就可以通過等待幾秒鐘來逐漸增加持續時間，然後再給予款待。這有助於加強 sit 命令。

Generalize the Command: Practice the "sit" command in different locations and with various distractions to help your dog generalize the behavior.

概括命令：在不同的位置和各種干擾下練習「坐下」命令，以説明您的狗概括行為。

Be Patient and Consistent: Patience and consistency are key in dog training. Always use positive reinforcement, and avoid punishment. If your dog doesn't succeed initially, go back a step and try again.

要有耐心和一致性： 耐心和一致性是狗訓練的關鍵。始終使用正強化，避免懲罰。如果您的狗最初沒有成功，請後退一步並重試。

Remember that each dog is unique, and some may learn more quickly than others. Adjust your training approach based on your dog's individual needs and progress.

請記住，每隻狗都是獨一無二的，有些狗可能比其他狗學得更快。根據狗的個人需求和進度調整您的訓練方法。

In [18]:
# chatgpt = BasicModelRunner("chat-gpt")

In [19]:
# print(chatgpt("Tell me how to train my dog to sit"))

status code: 561


APIError: API error {'detail': "Currently this user has support for base models: ['hf-internal-testing/tiny-random-gpt2', 'EleutherAI/pythia-70m', 'EleutherAI/pythia-70m-deduped', 'EleutherAI/pythia-70m-v0', 'EleutherAI/pythia-70m-deduped-v0', 'EleutherAI/neox-ckpt-pythia-70m-deduped-v0', 'EleutherAI/neox-ckpt-pythia-70m-v1', 'EleutherAI/neox-ckpt-pythia-70m-deduped-v1', 'EleutherAI/gpt-neo-125m', 'EleutherAI/pythia-160m', 'EleutherAI/pythia-160m-deduped', 'EleutherAI/pythia-160m-deduped-v0', 'EleutherAI/neox-ckpt-pythia-70m', 'EleutherAI/neox-ckpt-pythia-160m', 'EleutherAI/neox-ckpt-pythia-160m-deduped-v1', 'EleutherAI/pythia-2.8b', 'EleutherAI/pythia-410m', 'EleutherAI/pythia-410m-v0', 'EleutherAI/pythia-410m-deduped', 'EleutherAI/pythia-410m-deduped-v0', 'EleutherAI/neox-ckpt-pythia-410m', 'EleutherAI/neox-ckpt-pythia-410m-deduped-v1', 'cerebras/Cerebras-GPT-111M', 'cerebras/Cerebras-GPT-256M', 'meta-llama/Llama-2-7b-hf', 'meta-llama/Llama-2-7b-chat-hf', 'meta-llama/Llama-2-13b-chat-hf', 'meta-llama/Llama-2-70b-chat-hf', 'Intel/neural-chat-7b-v3-1', 'mistralai/Mistral-7B-Instruct-v0.1', 'microsoft/phi-2']. Need help? Email us at info@lamini.ai"}